src/Catalyst.Cli/CommandTypes/BaseMessageCommand.cs

```diff
@@ -0,0 +1,147 @@
+               .OfType<SocketClientRegistryClientAdded>().Subscribe(SocketClientRegistryClientAddedOnNext);
+            _eventStreamObserverClientRemoved = CommandContext.SocketClientRegistry.EventStream
+               .OfType<SocketClientRegistryClientRemoved>().Subscribe(SocketClientRegistryClientRemovedOnNext);
+        }
+
+        protected PeerId RecipientPeerId
+        {
+            get
+            {
+                if (_recipientPeerId != null) return _recipientPeerId;
+                var rpcClientConfig = CommandContext.GetNodeConfig(Options.Node);
+                _recipientPeerId =
+                    rpcClientConfig.PublicKey.BuildPeerIdFromBase32Key(rpcClientConfig.HostAddress,
+                        rpcClientConfig.Port);
+                return _recipientPeerId;
+            }
+        }
+
+        protected PeerId SenderPeerId => CommandContext.PeerId;
+
+        public void Dispose() { Dispose(true); }
+
+        public IRpcClient Target => CommandContext.GetConnectedNode(Options.Node);
+
+        public virtual void SendMessage(TOption options)
+        {
+            var message = GetMessage(options);
+
+            if (message == null) return;
+
+            var messageDto = new MessageDto(
+                message.ToProtocolMessage(SenderPeerId),
+                RecipientPeerId);
+            Target.SendMessage(messageDto);
+        }
+
+        protected abstract TRequest GetMessage(TOption option);
+
+        protected override bool ExecuteCommandInner(IOptionsBase optionsBase)
+        {
+            var sendMessage = base.ExecuteCommandInner(optionsBase);
+
+            if (sendMessage) SendMessage((TOption) optionsBase);
+
+            return sendMessage;
+        }
+
+        protected virtual void ResponseMessage(TResponse response)
+        {
+            CommandContext.UserOutput.WriteLine(response.ToJsonString());
+        }
+
+        private void CommandResponseOnNext(TResponse value) { ResponseMessage(value); }
+
+        private void SocketClientRegistryClientAddedOnNext(SocketClientRegistryClientAdded clientAddedEvent)
+        {
+            try
+            {
+                var client = CommandContext.SocketClientRegistry.GetClientFromRegistry(clientAddedEvent.SocketHashCode);
+                var subscription = client.SubscribeToResponse<TResponse>(CommandResponseOnNext);
+                _subscriptions.TryAdd(clientAddedEvent.SocketHashCode, subscription);
+            }
+            catch (Exception e)
+            {
+                Logger.Error(e, "Error subscribing to Rpc response stream.");
+            }
+        }
+
+        private void SocketClientRegistryClientRemovedOnNext(SocketClientRegistryClientRemoved clientRemovedEvent)
+        {
+            var subscriptionRemoved =
+                _subscriptions.TryRemove(clientRemovedEvent.SocketHashCode, out var outSubscription);
+            if (subscriptionRemoved) outSubscription.Dispose();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)